CiteProc.php

```diff
@@ -968,14 +968,20 @@ function render($data, $mode = NULL) {
     $text = '';
 
     if (($var = $this->variable) && isset($data->{$var})) {
-      $date = $data->{$var}->{'date-parts'}[0];
-      foreach ($this->elements as $element) {
-        $date_parts[] = $element->render($date, $mode);
+      if (isset($data->{$var}->{'date-parts'})) {
+        $date = $data->{$var}->{'date-parts'}[0];
+        foreach ($this->elements as $element) {
+          $date_parts[] = $element->render($date, $mode);
+        }
+        $text = implode('', $date_parts);
+        return $this->format($text);
+      }
+      else if (isset($data->{$var}->{'literal'})) {
+        return $this->format($text);
       }
-      $text = implode('', $date_parts);
     }
 
-    return $this->format($text);
+    return $this->citeproc->get_locale('term', 'no date');
```

Comment:
It seems that if I leave Issued blank this class is not called at all. So this does not generate the no date term if it is added to the CSL.